### PR TITLE
chore: Check that PR title matches conventional commit guidelines

### PR DESCRIPTION
**Pull request by kdheepak**
This runs a quick check to see if the PR title matches conventions commit guidelines. This action reruns if the title is edited and modified.

### CODE REVIEW

**Comment by a-kenji** [comment]
I assumed the `committed` check does that already. Is that incorrect?

**Review by mindoodoo — APPROVED** [review]
@a-kenji Unfortunately not :/

Thanks kd !

File: .github/workflows/check-pr-title.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      - edited
+
+jobs:
+  check-title:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check PR title
+      run: |
+        title="${{ github.event.pull_request.title }}"
+        if ! echo "$title" | grep -qE '^(build|chore|ci|docs|style|refactor|perf|test|feat|fix): '; then
```
Comment (joshka) [inline .github/workflows/check-pr-title.yml:18]:
```suggestion
        if ! echo "$title" | grep -qE '^(build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test)(\(\w+\))?!?: '; then
```
Adds:
- `revert` (from committed.toml)
- optional scope: `(scope)`
- optional `!` to indicate breaking changes

**Review by joshka — CHANGES_REQUESTED** [review]

**Review by joshka — APPROVED** [review]

**Comment by joshka** [comment]
Let's squash this and then merge.

**Pull request merged**